demo/scripts/controls/ribbonButtons/contentModel/spaceBeforeAfterButtons.ts

```diff
@@ -0,0 +1,47 @@
+import { RibbonButton } from 'roosterjs-react';
+import { isContentModelEditor, setParagraphMargin } from 'roosterjs-content-model';
+
+const spaceAfterButtonKey = 'buttonNameSpaceAfter';
+const spaceBeforeButtonKey = 'buttonNameSpaceBefore';
+
+/**
+ * @internal
+ * "Add space after" button on the format ribbon
+ */
+export const spaceAfterButton: RibbonButton<typeof spaceAfterButtonKey> = {
+    key: spaceAfterButtonKey,
+    unlocalizedText: 'Remove space after',
+    iconName: 'CaretDown8',
+    isChecked: formatState => !formatState.marginBottom || parseInt(formatState.marginBottom) <= 0,
+    onClick: (editor, _, _1, _2, { marginBottom }) => {
+        if (isContentModelEditor(editor)) {
+            setParagraphMargin(
+                editor,
+                undefined /* marginTop */,
+                parseInt(marginBottom) ? null : '8pt'
+            );
+        }
+        return true;
+    },
+};
+
+/**
+ * @internal
+ * "Add space before" button on the format ribbon
+ */
+export const spaceBeforeButton: RibbonButton<typeof spaceBeforeButtonKey> = {
+    key: spaceBeforeButtonKey,
+    unlocalizedText: 'Add space before',
+    iconName: 'CaretUp8',
+    isChecked: formatState => parseInt(formatState.marginTop) > 0,
+    onClick: (editor, _, _1, _2, { marginTop }) => {
+        if (isContentModelEditor(editor)) {
+            setParagraphMargin(
```

Comment:
If you want to do a toggle, you can get format state using getFormatState() here. Do not let plugin pass the format to you. Or if you really want, make it an optional parameter so that we know it can be null.